FileNotFoundError when downloading PDF volumes

### What happened ?

After downloading the chapters when the worker converts the volume into a PDF I get a FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Ethan\\Downloads\\Kobo-pine\\New Normal\\Vol. 3\\000.png

Full error below:
```
[ERROR] We have problem in queue worker
Traceback (most recent call last):
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\utils.py", line 217, in run
    job()
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 454, in <lambda>
    self.worker.submit(lambda: self.convert(self.images, file_path))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 392, in convert
    im = im_ref()
         ^^^^^^^^
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 68, in __call__
    return self._func(*self._args, **self._kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\PIL\Image.py", line 3236, in open
    fp = builtins.open(filename, "rb")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Ethan\\Downloads\\Kobo-pine\\New Normal\\Vol. 3\\000.png'
[ERROR] Unhandled exception, FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Ethan\\Downloads\\Kobo-pine\\New Normal\\Vol. 3\\000.png'
Traceback (most recent call last):
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\cli\__init__.py", line 76, in _main
    download(args)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\cli\download.py", line 33, in download
    url(args, args.type)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\cli\url.py", line 166, in __call__
    self.func(self.id, args)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\cli\url.py", line 97, in download_manga
    dl_manga(*args)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\main.py", line 167, in download
    download_manga(manga, base_path)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\main.py", line 137, in download_manga
    fmt.main()
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\base.py", line 697, in main
    self.download_volumes(volumes)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\base.py", line 625, in download_volumes     
    self.on_finish(file_path, volume, images)
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 454, in on_finish
    self.worker.submit(lambda: self.convert(self.images, file_path))
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\utils.py", line 196, in submit
    raise err
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\utils.py", line 217, in run
    job()
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 454, in <lambda>
    self.worker.submit(lambda: self.convert(self.images, file_path))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 392, in convert
    im = im_ref()
         ^^^^^^^^
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\mangadex_downloader\format\pdf.py", line 68, in __call__
    return self._func(*self._args, **self._kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\AppData\Local\Programs\Python\Python311\Lib\site-packages\PIL\Image.py", line 3236, in open
    fp = builtins.open(filename, "rb")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Ethan\\Downloads\\Kobo-pine\\New Normal\\Vol. 3\\000.png'
```

### What did you expect to happen ?

I expected the program to convert the Folders into a PDF, in the same way as the cbz method

### OS version

Windows 10

### App version

mangadex-downloader v2.10.1 (https://github.com/mansuf/mangadex-downloader)
Python: 3.11.2
arch: x64
bundled executable: no

### Installation origin

PyPI (Python Package Index)

### Installation origin (other sources)

_No response_

### Reproducible command

mangadex-dl https://mangadex.org/title/e1d0056a-fdd3-4f32-af19-50eeb37280ac --save-as "pdf-volume" --use-volume-cover -uvc --progress-bar-layout=none

### Additional context

if you repeatedly do the same command over and over every time it crashes it will eventually get through all the chapters (using the download.db I'm guessing)

--- a/docs/changelog.md
+++ b/docs/changelog.md
@@ -1,5 +1,12 @@
 # Changelog
 
+## UNRELEASED
+
+### Fix bugs
+
+- Fixed `pdf-volume` and `cb7-volume` formats are not working after upgrading to v2.10.0 
+[#78](https://github.com/mansuf/mangadex-downloader/issues/78)
+
 ## v2.10.1
 
 ### Fix bugs

--- a/mangadex_downloader/format/base.py
+++ b/mangadex_downloader/format/base.py
@@ -574,8 +574,8 @@ class ConvertedVolumesFormat(BaseConvertedFormat):
         """This function is called when format has successfully received images"""
         pass
 
-    def on_finish(self, file_path, volume, images):
-        """"This function is called after download is finished"""
+    def on_convert(self, file_path, volume, images):
+        """"This function is called when convert process is starting"""
         pass
 
     def download_volumes(self, volumes):
@@ -622,7 +622,7 @@ class ConvertedVolumesFormat(BaseConvertedFormat):
                 pbm.get_pages_pb().reset()
 
             chapters_pb.reset()
-            self.on_finish(file_path, volume, images)
+            self.on_convert(file_path, volume, images)
                 
             # Remove original chapter folder
             shutil.rmtree(volume_path, ignore_errors=True)

--- a/mangadex_downloader/format/comic_book.py
+++ b/mangadex_downloader/format/comic_book.py
@@ -195,7 +195,7 @@ class ComicBookArchiveVolume(ConvertedVolumesFormat, CBZFile):
     def on_iter_chapter(self, file_path, chapter, count):
         self.insert_ch_info_img(self.volume_zip, self.worker, chapter, count, self.volume_path)
 
-    def on_finish(self, file_path, volume, images):
+    def on_convert(self, file_path, volume, images):
         volume_name = self.get_volume_name(volume)
         pbm.logger.info(f"{volume_name} has finished download, converting to cbz...")
 

--- a/mangadex_downloader/format/epub.py
+++ b/mangadex_downloader/format/epub.py
@@ -435,7 +435,7 @@ class EpubVolume(ConvertedVolumesFormat, EPUBFile):
     def on_prepare(self, file_path, volume, count):
         self.epub_chapters = []
 
-    def on_finish(self, file_path, volume, images):
+    def on_convert(self, file_path, volume, images):
         volume_name = self.get_volume_name(volume)
 
         pbm.logger.info(f"{volume_name} has finished download, converting to epub...")

--- a/mangadex_downloader/format/pdf.py
+++ b/mangadex_downloader/format/pdf.py
@@ -440,6 +440,9 @@ class PDFVolume(ConvertedVolumesFormat, PDFFile):
         self.images = []
 
     def on_prepare(self, file_path, volume, count):
+        # We should clear self.images, to prevent older volumes to be re-converted
+        self.images.clear()
+
         volume_name = self.get_volume_name(volume)
         self.volume_path = create_directory(volume_name, self.path)
 
@@ -448,7 +451,7 @@ class PDFVolume(ConvertedVolumesFormat, PDFFile):
     def on_iter_chapter(self, file_path, chapter, count):
         self.insert_ch_info_img(self.images, chapter, self.volume_path, count)
 
-    def on_finish(self, file_path, volume, images):
+    def on_convert(self, file_path, volume, images):
         volume_name = self.get_volume_name(volume)
         pbm.logger.info(f"{volume_name} has finished download, converting to pdf...")
         self.worker.submit(lambda: self.convert(self.images, file_path))

--- a/mangadex_downloader/format/sevenzip.py
+++ b/mangadex_downloader/format/sevenzip.py
@@ -112,6 +112,8 @@ class SevenZipVolume(ConvertedVolumesFormat, SevenZipFile):
         self.images = []
 
     def on_prepare(self, file_path, volume, count):
+        self.images.clear()
+
         volume_name = self.get_volume_name(volume)
         self.volume_path = create_directory(volume_name, self.path)
 
@@ -123,7 +125,7 @@ class SevenZipVolume(ConvertedVolumesFormat, SevenZipFile):
     def on_received_images(self, file_path, chapter, images):
         self.images.extend(images)
 
-    def on_finish(self, file_path, volume, images):
+    def on_convert(self, file_path, volume, images):
         volume_name = self.get_volume_name(volume)
 
         pbm.logger.info(f"{volume_name} has finished download, converting to cb7...")

